src/WooCommerce.php

```diff
@@ -14,14 +14,16 @@ class WooCommerce {
 
   /**
    * Adds data-srcset attributes to image links to make them reponsive in lightGallery.
+   *    * Adds data-sizes attributes to image links so they don't get enlarged more than the original size.
    *
    * @see https://sachinchoolur.github.io/lightGallery/demos/responsive.html
    *
    * @implements woocommerce_single_product_image_thumbnail_html
    */
   public static function woocommerce_single_product_image_thumbnail_html($html, $thumbnail_id) {
     $srcset = wp_get_attachment_image_srcset($thumbnail_id, 'shop_single');
-    return preg_replace('/(<a\s+)/i', '<a data-srcset="' . $srcset . '" ', $html);
+    $srcsizes = wp_get_attachment_image_sizes($thumbnail_id, 'large');
```

Comment:
Alright @sun, I have understood what I did now ^^

 I'll change it to use the 'full' size, since the 'large' one will display a maximum 1024px width and height. I didn't notice a difference because all the uploaded images I tested are not larger than 1000px, but now I see, the correct size would be 'full', so we keep the original size of each image not limiting its display.